python/triton/backends/compiler.py

```diff
@@ -1,12 +1,21 @@
-from abc import ABCMeta, abstractmethod, abstractclassmethod
 import os
-import subprocess
 import re
+import subprocess
+
+from abc import ABCMeta, abstractmethod, abstractclassmethod
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class GPUTarget(object):
```

Comment:
I can see the value of using a dataclass instead of a tuple to describe the target info. Now we can see the exact fields. 
However, we already have HIPBackend and CUDABackend. Under these backends, we have this GPUTarget, which further has target and backend.
Maybe we can introduce new terms here?
- "GPGPU" to replace the backend here. It could be either cuda or hip. (suggested by chatgpt)
- "arch" to replace target. it could be either gfxyyy on hip or the computeCapability on cuda. If gfxyyy and computeCapability cannot be merged easily, is it possible to keep both in this GPUTarget and let backend decide which one to use?

Reply:
Thanks for the suggestion! Using `arch` to replace `target` sounds good to me; so changed that. I'm not so sure about `GPGPU` though--it's somewhat historical concepts that emphasizes the time back when GPUs are more meant for graphics. We are quite past that time and in the scope of Triton (and AI) it's pretty clear that GPU is just for compute. Also `GPGPU` is a fairly broad and abstract term too. A more suitable one might be `execution_environment` but it's quite lenghty. Or maybe `execution_api` or just `api`. I'm a bit on the fence about them too though. We can continue discussing about this. For now I'll just keep `backend`.

Reply:
For sure. I'm not sure about GPGPU either. Just want to suggest something to initiate the discussion. 
But I think adding comments here helps a lot. We should be good for now.